models/Category.php

```diff
@@ -256,7 +256,11 @@ public static function resolveMenuItem($item, $url, $theme)
                 'items' => []
             ];
 
-            $categories = self::orderBy('name')->get();
+            $categories = self::select(self::query()->qualifyColumn('*'))
+                ->leftJoin(self::posts()->getTable(), self::posts()->getQualifiedParentKeyName(), '=', self::posts()->getQualifiedForeignPivotKeyName())
+                ->leftJoin(self::posts()->getRelated()->getTable(), self::posts()->qualifyColumn(self::posts()->getRelatedKeyName()), '=', self::posts()->getQualifiedRelatedPivotKeyName())
+                ->where(\Closure::fromCallable([new Post, 'scopeIsPublished']))->orderBy('name')->get();
```

Comment:
I'm not sure this extra complexity is worth it, the case that it is implemented for seems a bit edge case to me (500 separate categories? What on earth are you using your blog posts for?)

Reply:
@LukeTowers I have a site with 80 categories (nested in several levels, of course). It's not that far-fetched - especially for big news sites.

Reply:
@zaxbux could this join cluster be converted to a scope? It might be a bit more palatable that way :P

Reply:
Agreed, 500 categories is more of an edge case, although I chose those numbers for testing potential performance issues.